wodles/aws/aws_s3.py

```diff
@@ -171,7 +171,21 @@ def main(argv):
                 )
             elif options.subscriber.lower() == "buckets":
                 bucket_handler = subscribers.s3_log_handler.AWSSubscriberBucket
-                asl_queue = subscribers.sqs_queue.AWSSQSQueue(
+                subscriber_queue = subscribers.sqs_queue.AWSSQSQueue(
+                    iam_role_arn=options.iam_role_arn,
+                    iam_role_duration=options.iam_role_duration,
+                    profile=options.aws_profile,
+                    sts_endpoint=options.sts_endpoint,
+                    service_endpoint=options.service_endpoint,
+                    name=options.queue,
+                    skip_on_error=options.skip_on_error,
+                    discard_field=options.discard_field,
+                    discard_regex=options.discard_regex,
+                    bucket_handler=bucket_handler,
+                    message_processor=subscribers.sqs_message_processor.AWSS3MessageProcessor)
+            elif options.subscriber.lower() == "security_hub":
+                bucket_handler = subscribers.s3_log_handler.AWSSecurityHubSubscriberBucket
+                subscriber_queue = subscribers.sqs_queue.AWSSQSQueue(
```

Comment:
Since the `buckets` and `security_hub` subscriber options share the same parameters, we could consider instantiating similarly as for services or buckets to repeat less code (ideally, also for ASL):
https://github.com/wazuh/wazuh/blob/098c3622e15581f3fdf2ce947a41907edeb4d8e5/wodles/aws/aws_s3.py#L52-L98

It's not important and may be more complicated than it seems initially, so take this as a suggestion, not mandatory!

Reply:
Carried out the requested modification and checked that did not break the current functionalities:
- The `"security_lake"` case keeps its validations to raise errors if any of the authentication parameters are incorrect or empty and the instantiation now receives the `discard_field` and `discard_regex` parameters. These parameters are considered if they are defined in the `ossec.conf` breaking the execution, and the only way to pass them to the script is to execute it manually, which would not influence the operation since when the parquet files are processed, they are not taken into account.
- For the `buckets` and `security_hub` cases, the `external_id` is now passed but since its default value is `None` unless a user defines it, it does not modify the behavior.